src/server/snapshot.cc

```diff
@@ -242,7 +242,8 @@ bool SliceSnapshot::BucketSaveCb(PrimeIterator it) {
     ++stats_.skipped;
     return false;
   }
-  db_slice_->FlushChangeToEarlierCallbacks(current_db_, it, snapshot_version_);
+  db_slice_->FlushChangeToEarlierCallbacks(current_db_, DbSlice::Iterator::FromPrime(it),
```

Comment:
why do we need to call DbSlice::Iterator::FromPrime here?

Reply:
`FlushChangeToEarlierCallbacks` calls pre-update callbacks. I _think_ (but I'm not sure) that we will need to give them the auto-laundering iterator as well, as they might also yield.
wdyt?

Reply:
is not the whole motivation of this PR is to allow `SliceSnapshot::OnDbChange` to yield?

Reply:
We definitely want OnDbChange to be able to yield. And I think it's a good idea that it will have an auto laundering iterator to make things easier there. But it's not a necessity as the callbacks could launder automatically and anyway they should be using the values and not find etc.
but again, I think it's a good idea.